Move SDPA decomp pass from Qualcomm's directory to be shareable and call it for Cadence backends

mcremon-meta opened the pull request:
Summary: Moving the pass to backends/transforms so that other backends can call it, and call it from the Cadence side so that we can quantize the bmm ops in SDPA.

Differential Revision: D59600486

cccclai commented on backends/transforms/decompose_sdpa.py:
```diff
@@ -1,8 +1,11 @@
-# Copyright (c) Qualcomm Innovation Center, Inc.
-# All rights reserved
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
For Qualcomm folks (@shewu-quic , @chunit-quic, @haowhsu-quic) since the decompose sdpa is helpful for other backends, mind us moving it to a central place so we can reuse?

haowhsu-quic replied:
Sure, that's fine. Good to see this!

mcremon-meta committed:
Move SDPA decomp pass from Qualcomm's directory to be shareable and call it for Cadence backends (#4258)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4258

Moving the pass to backends/transforms so that other backends can call it, and call it from the Cadence side so that we can quantize the bmm ops in SDPA.

Reviewed By: cccclai

Differential Revision: D59600486
